integrationtests/src/test/java/io/kroxylicious/proxy/ApiVersionsIT.java

```diff
@@ -75,9 +76,8 @@ public void shouldOfferTheMaximumLowestSupportedVersionWhenKroxyliciousIsAheadOf
         }
     }
 
-    // TODO we think this is a bug but will come back to it
     @Test
-    public void shouldOfferBrokerApisThatAreUnknownToKroxy() {
+    public void shouldNotOfferBrokerApisThatAreUnknownToKroxy() {
```

Comment:
The test would pass if the production code had a bug that dropped all api keys from the response in the presence of a single unknown one.

I think the test would be stronger if it added two apiKeys (line 86), one known, and one unknown.  The test would then assert that only the known one makes it to the client.

Reply:
Getting back on my "one reason to fail" soap box. I think @k-wall is making a great point *but* it's two separate tests. 

Tests:
1. `shouldNotOfferBrokerApisThatAreUnknownToKroxy` should assert that unknown API's are removed (as it does).
2. `shouldOfferBrokerApisThatAreKnownToKroxy` is the extension behaviour @k-wall outlines. 

I say this because if we only had test 2. and it fails in 6 months time, we don't know what went wrong we just know something changed in the filtering as there are two possible causes. 1. We stopped filtering an unknown API (maybe the API key stopped being unknown, maybe we broke it) or 2. We stopped including the known APIs. 

So for me the value of having two tests comes later. 



Now all this comes with the caveat of being slower to execute (we have to spool everything up twice), for the moment I'd argue we are not that sensitive to the build time and thus can afford it. There is also an argument about duplication but I think we already have the right abstractions in place to make that minimal and naming the test cases well (that is to say giving each one a meaningful name) mitigates that by showing the difference in intent.

Reply:
I've pushed up the inverse test `shouldOfferBrokerApisThatAreKnownToKroxy` to check all known ApiKeys can be represented.